add output/export/xlsx_IIASA.R that generates IAMC submission file

orichters opened the pull request:
## Purpose of this PR

- the process of generating files for the submission to the IIASA database is quite tedious, requiring copying files back and forth to project interfaces etc.
- with this PR, a new option `xlsx_IIASA` will be available in `output.R` for `export`. In the script, you can specify the mapping file generated with [`project_interfaces`](https://github.com/pik-piam/project_interfaces/) and the `iiasatemplate` containing the list of variables allowed in the database, specified in each project (link to xlsx or yaml file)
- `output.R` asks you to specify the output subfolders you want to use plus a filename_prefix such as `NGFS`
- additional options can be specified, either by editing the `xlsx_IIASA.R` file or via the command line starting `Rscript output.R LOGFILE=mylogfile.txt`, or `project=NGFS` which loads project-specific settings
- As a result, you get an `.xlsx` and a `.mif` file using the mapping you required (filename will then be `REMIND-_NGFS_timestamp.xlsx`), and file with `_remind2.mif` at the end which contains all the original mif data from the scenarios, and a logfile with additional infos about non-used variable names
- The script is essentially a collection of code from https://github.com/pik-piam/project_interfaces/blob/master/produce_output.R, https://github.com/pik-piam/project_interfaces/blob/master/ar6/fixVariableNames_removeNAs.sh and some code Jérôme and I wrote for NGFS

## Type of change

- [x] New feature, no impact on scenarios

## Checklist:

- [x] My code follows the coding etiquette
- [x] I have performed a self-review of my own code
- [x] Changes are commented, particularly in hard-to-understand areas
- [x] The model compiles and runs successfully (`Rscript start.R -q`)

orichters commented:
>     * start by renaming the script for something more general like "project_report"

I'm completely open to suggestions.

>     * and set whatever code is specific to your mapping under a conditional, i.e., from a quick look to the code it appears that there are some units renaming and other code lines inside of the script that are specific for your project/report case.

The [unit renaming](https://github.com/remindmodel/remind/pull/909/files#diff-d73383775e8e39a227201a0870b029b6198fcab3be55e22fb65537cd7621e771R147-R152) is something that is (if I understood IIASA correctly) merely an error correction for units that were inconsistently spelled in the AR6 template. All future submissions to the IIASA database will need this unit correction. The rest is to be quite generic, although I am not sure whether [these options](https://github.com/remindmodel/remind/pull/909/files#diff-d73383775e8e39a227201a0870b029b6198fcab3be55e22fb65537cd7621e771R123-R140) are always required/desirable.

> I wouldn't say this generalization need to be included in your initial commit, but if after discussing with RSE it is decided that this will be the way to go to create project reports, I am afraid that in the not so long term the script will be useless and outdated if you have hard coded transformations directly in the code. Besides it would be a loss opportunity and useless for other projects and updated mappings.

I'm completely open if somebody wants to rewrite the script, put it to `remind2` or whatever. And I don't think it will be outdated quickly, the scripts in `project_interfaces` seemed to be quite stable in the past. Of course, the files specified as `MAPPING` and `iiasatemplate` will change every time, but that is process-specific anyhow.

Renato-Rodrigues commented:
> But: what other databases exist we submit to? At least the README of [project_interfaces](https://github.com/pik-piam/project_interfaces/) talks only about IIASA.

Project templates can be also used in standalone products not related with iiasa in any way, like webpages showing specific results (e.g. INNOPATHS used this). Besides, there are different mappings within the iiasa db for different projects. They can simply extend the variable templates, or include changes to the variables naming convention to be better comparable with other bottom-up model results or project specific needs. There is no need for this to be an iiasa only feature in this sense.

> We may also think about adding a project variable that can be used to switch on and off project-specific code.

That is exactly what I had in mind.

Renato-Rodrigues commented:
> Ok, but then somebody with specific knowledge of these projects should check whether this code is somehow reusable. 

Yes, that is why I said this should not hold your commit back. I would only change the name of the script to something more general like "project_report" or something similar and commit the way it is now. 

>> Of course, but this is captured in the `MAPPING` and `iiasa_template` variable anyhow that users have to specify (and provide the files) before they can use the scripts.

Yes, that is true as long as there is no hard coded changes in the "xlsx_iiasa" script for a given template that could error out while using other templates as I mentioned above.

orichters commented:
Maybe instead of having one single scripts with lots of options:
- introduce a new category/folder `project_reports` additional to `single` and `comparison`, and have multiple scripts for different projects there: `xlsx_IIASA`, `INNOPATHS`, `SHAPE` etc.?

Loisel commented on scripts/output/comparison/xlsx_IIASA.R:
```diff
@@ -0,0 +1,214 @@
+
+options(warn = 1)
+
+MODEL <- "REMIND-MAgPIE 3.0-4.4"                            # modelname in final file
```
maybe not worth the effort, especially with the high temperatures.

orichters replied:
And I think we can't, because this depends on what model name is registered with IIASA.

orichters commented on output.R:
```diff
@@ -214,7 +214,7 @@ if (comp == "Exit") {
 
   # ask for filename_prefix, if one of the modules that use it is selected
-  modules_using_filename_prefix <- c("compareScenarios", "compareScenarios2")
+  modules_using_filename_prefix <- c("compareScenarios", "compareScenarios2", "xlsx_IIASA")
```
Although this is not the question here, in this code line, this is just all the comparison modules that let the user specify a prefix.

orichters commented:
Ok, I added a new category to `output.R`. I opted for `export` because this describes more clearly what I expect to end in this category: Generate files that are then used as input to other models, databases etc. I hope that is fine (and shorter than `project_reports` as well).

I added a command-line argument `project` and put all the unit changes into a conditional that `project` is either `NGFS` or `ENGAGE4p5`. If other people want to use it as well, they can add their project name quickly, but the default is that no units are changed.
